microsoft_auth/apps.py

```diff
@@ -1,139 +1,51 @@
-import importlib
+import base64
+import logging
 
-from django.apps import AppConfig, apps
-from django.core.checks import Critical, Warning, register
-from django.db.utils import OperationalError, ProgrammingError
-from django.test import RequestFactory
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes, serialization
+from django.apps import AppConfig
 
-
-class MicrosoftAuthConfig(AppConfig):
-    name = "microsoft_auth"
-    verbose_name = "Microsoft Auth"
+logger = logging.getLogger("django")
 
 
-@register()
-def microsoft_auth_validator(app_configs, **kwargs):
-    from django.contrib.sites.models import Site
-
-    from .conf import HOOK_SETTINGS, config
+def slurp_file(filename, file_type):
+    try:
+        file = open(filename, "r").read()
+    except IOError:
+        logger.error("Cannot open or read file: {}".format(filename))
+        file = None
+    return file
 
-    errors = []
 
-    if apps.is_installed("microsoft_auth") and not apps.is_installed(
-        "django.contrib.sites"
-    ):
+class MicrosoftAuthConfig(AppConfig):
+    name = "microsoft_auth"
+    verbose_name = "Microsoft Auth"
 
-        errors.append(
-            Critical(
-                "`django.contrib.sites` is not installed",
-                hint=(
-                    "`microsoft_auth` requires `django.contrib.sites` "
-                    "to be installed and configured"
-                ),
-                id="microsoft_auth.E001",
-            )
-        )
+    def ready(self):
+        from . import checks  # noqa
+        from .conf import config
 
-    try:
-        if not hasattr(config, "SITE_ID"):
-            request = RequestFactory().get("/", HTTP_HOST="example.com")
-            current_site = Site.objects.get_current(request)
-        else:
-            current_site = Site.objects.get_current()
-    except Site.DoesNotExist:
-        pass
-    except (OperationalError, ProgrammingError):
-        errors.append(
-            Warning(
-                "`django.contrib.sites` migrations not ran",
-                id="microsoft_auth.W001",
-            )
-        )
-    else:
-        if current_site.domain == "example.com":
-            errors.append(
-                Warning(
-                    (
-                        "`example.com` is still a valid site, Microsoft "
-                        "auth might not work"
-                    ),
-                    hint=(
-                        "Microsoft/Xbox auth uses OAuth, which requires "
-                        "a real redirect URI to come back to"
-                    ),
-                    id="microsoft_auth.W002",
-                )
-            )
+        if config.MICROSOFT_AUTH_ASSERTION_CERTIFICATE != "":
+            cert = config.MICROSOFT_AUTH_ASSERTION_CERTIFICATE
```

Comment:
Can you try renaming this variable to `cert_filename` or something to see if we can make that CodeQL error go away.

It is not actually logging sensitive data, so if we cannot get it to go away, I can just dismiss it, but if it is not fixed, it will keep coming back.

Reply:
That fixed it :)